federation/pkg/federation-controller/replicaset/replicasetcontroller.go

```diff
@@ -485,7 +463,6 @@ func (frsc *ReplicaSetController) reconcileReplicaSet(key string) (reconciliatio
 	if err != nil {
 		glog.Errorf("Failed to ensure delete object from underlying clusters finalizer in replicaset %s: %v",
 			frs.Name, err)
-		frsc.deliverReplicaSetByKey(key, 0, false)
```

Comment:
Same. There are lots of places in this method where we call deliverReplicaSetByKey before returning statusError.
You want to remove all them?